x-pack/plugins/endpoint/server/routes/metadata/query_builders.ts

```diff
@@ -46,37 +49,43 @@ export const kibanaRequestToMetadataListESQuery = async (
 };
 
 async function getPagingProperties(
-  request: KibanaRequest<any, any, any>,
+  request: KibanaRequest<unknown, unknown, MetadataIndexGetBodyResult>,
   endpointAppContext: EndpointAppContext
 ) {
   const config = await endpointAppContext.config();
   const pagingProperties: { page_size?: number; page_index?: number } = {};
-  if (request?.body?.paging_properties) {
+  if (request.body?.paging_properties) {
     for (const property of request.body.paging_properties) {
-      Object.assign(
-        pagingProperties,
-        ...Object.keys(property).map(key => ({ [key]: property[key] }))
-      );
+      if ('page_size' in property) {
```

Comment:
working on the typed values here. Out of curiosity, why is the API designed to take an array of objects that have either page_size or page_index? why not just `{ page_size?: number, page_index?: number}`?